express-crud/server.ts

```diff
@@ -0,0 +1,139 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/grades', async (req, res, next) => {
+  try {
+    const sql = `
+    select *
+    from "grades";
+    `;
+    const result = await db.query(sql);
+    res.status(200).json(result.rows);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/grades:gradeId', async (req, res, next) => {
+  try {
+    const { gradeId } = req.params;
+    if (gradeId === undefined) {
```

Comment:
It's best to check `if (!Number.isInteger(+gradeId))` so you can make sure they sent you an integer.